salt/modules/selinux.py

```diff
@@ -240,16 +240,36 @@ def list_semod():
 
     .. versionadded:: 2016.3.0
     '''
-    mdata = __salt__['cmd.run']('semodule -l').splitlines()
-    ret = {}
-    for line in mdata[1:]:
-        if not line.strip():
-            continue
-        comps = line.split()
-        if len(comps) == 3:
-            ret[comps[0]] = {'Enabled': False,
-                             'Version': comps[1]}
-        else:
-            ret[comps[0]] = {'Enabled': True,
-                             'Version': comps[1]}
+    semodule_versioncheck = __salt__['cmd.run']('semodule -h').splitlines()
+    semodule_version = ''
+    for line in versioncheck[1:]:
+        if line.strip().startswith('full'):
+            semodule_version = 'new'
+
+    if semodule_version == 'new':
+        mdata = __salt__['cmd.run']('semodule -lfull').splitlines()
+        ret = {}
+        for line in mdata[0:]:
```

Comment:
I was merely parroting what was there before (well, it was mdata[1:] for some reason, but I digress).